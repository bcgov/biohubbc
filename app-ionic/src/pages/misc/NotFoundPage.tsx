import React from 'react';
import { Container, Grid } from '@material-ui/core';
import { Link } from 'react-router-dom';

export const NotFoundPage = () => {
  return (
    <Container>
      <Grid container direction="row">
        <Grid item>
          <h1>Page not found</h1>
          <Link to="/home/activities">Go back to the activities page</Link>
        </Grid>
      </Grid>
    </Container>
  );
};
