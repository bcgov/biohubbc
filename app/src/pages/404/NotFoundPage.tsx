import { Container, Grid } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => {
  return (
    <Container>
      <Grid container direction="row">
        <Grid item>
          <h1>Page not found</h1>
          <Link to="/home">Go back to the home page</Link>
        </Grid>
      </Grid>
    </Container>
  );
};

export default NotFoundPage;
