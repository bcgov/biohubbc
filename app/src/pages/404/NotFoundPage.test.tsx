import { fireEvent, render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import NotFoundPage from './NotFoundPage';

const history = createMemoryHistory();

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Router history={history}>
        <NotFoundPage />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('takes the user home when they click the return home button', () => {
    const { getByText } = render(
      <Router history={history}>
        <NotFoundPage />
      </Router>
    );

    fireEvent.click(getByText('Return Home'));

    expect(history.location.pathname).toEqual('/');
  });
});
