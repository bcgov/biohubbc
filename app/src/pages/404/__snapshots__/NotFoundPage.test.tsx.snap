// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <h1>
        Page not found
      </h1>
      <a
        href="/home"
        onClick={[Function]}
      >
        Go back to the home page
      </a>
    </div>
  </div>
</div>
`;
